lib/src/main/java/graphql/nadel/NextgenEngine.kt

```diff
@@ -408,39 +414,49 @@ internal class NextgenEngine(
             )
         }
 
-        val transformedData: MutableJsonMap = serviceExecResult.data.let { data ->
-            data.takeIf { transformedQuery.resultKey in data }
-                ?: mutableMapOf(transformedQuery.resultKey to null)
-        }
+        val transformedData: MutableJsonMap = serviceExecResult.data
```

Comment:
nit, why can't we use immutable approach like

serviceExecResult.data.filter((k,v) -> topLevelFields.keySet.contain(k)).toMutableMap